k8srm-prototype/pkg/api/claim_types.go

```diff
@@ -355,16 +298,76 @@ type DeviceConfigReference struct {
 	Name string `json:"name"`
 }
 
-// DevicePoolAllocation contains the pool and number of selected devices.
-type DevicePoolAllocation struct {
+// DeviceAllocation contains an individual device allocation result, including
+// per-device resource allocations, when applicable.
+type DeviceAllocation struct {
+	// DevicePoolName is the name of the DevicePool to which this
+	// device belongs.
+	// +required
+	DevicePoolName string `json:"devicePoolName"`
 
-	// DevicePoolName contains the name of the DevicePool for this
-	// allocation.
+	// DeviceName contains the name of the allocated Device.
 	// +required
-	DevicePoolName string `json:"devicePoolName,omitempty"`
+	DeviceName string `json:"deviceName,omitempty"`
 
-	// DeviceCount contains the number of devices allocated from the
-	// pool to satisfy this claim.
+	// Allocations contain the resource allocations from this device,
+	// for the claim. Note that this may only satisfy part of the claim.
+	// Also, because devices may allocate some resources in blocks, this
+	// may even be larger than the requests or limits in the claim.
+	//
+	// +optional
+	// +listType=atomic
+	Allocations []ResourceAllocation `json:"allocations,omitempty"`
+}
+
+// ResourceAllocation contains the per-device resource allocations.
+type ResourceAllocation struct {
+	// Name is the resource name/string for this allocation.
 	// +required
-	DeviceCount int `json:"deviceCount,omitempty"`
+	Name string `json:"name"`
+
+	// Amount is the amount of resource allocated for this claim.
+	// +required
+	Allocation resource.Quantity `json:"allocation,omitempty"`
+}
+
+type DeviceIP struct {
+	// IP is the IP address assigned to the device
+	IP string `json:"ip,omitempty"`
+}
+
+// AllocatedDeviceStatus contains the status of an allocated device, if the
+// driver chooses to report it. This may include driver-specific information.
+type AllocatedDeviceStatus struct {
+	// DevicePoolName is the name of the DevicePool to which this
+	// device belongs. The driver for that device pool owns this
+	// entry.
+	//
+	// +required
+	DevicePoolName string `json:"devicePoolName"`
+
+	// DeviceName contains the name of the allocated Device.
+	//
+	// +required
+	DeviceName string `json:"deviceName,omitempty"`
+
+	// Conditions contains the latest observation of the device's state.
```

Comment:
What I am going for here is the status of the claim on the device. A claim may apply device configuration, for example. That may fail. I was thinking that would appear here rather than in DevicePool.DeviceStatuses. WDYT?